gapii/cc/spy.cpp

```diff
@@ -238,10 +247,61 @@ EGLBoolean Spy::eglInitialize(CallObserver* observer, EGLDisplay dpy, EGLint* ma
 
 EGLContext Spy::eglCreateContext(CallObserver* observer, EGLDisplay display, EGLConfig config,
                                  EGLContext share_context, EGLint* attrib_list) {
-    EGLContext res = GlesSpy::eglCreateContext(observer, display, config, share_context, attrib_list);
-    for (int i = 0; attrib_list != nullptr && attrib_list[i] != EGL_NONE; i += 2) {
-        if (attrib_list[i] == EGL_CONTEXT_MAJOR_VERSION_KHR) {
-            GAPID_INFO("eglCreateContext requested GL version %i", attrib_list[i+1]);
+    // Read attrib list
+    std::map<EGLint, EGLint> attribs;
+    while(attrib_list != nullptr && *attrib_list != EGL_NONE) {
+        EGLint key = *(attrib_list++);
+        EGLint val = *(attrib_list++);
+        attribs[key] = val;
+    }
+
+    // Modify attrib list
+    if (mRecordGLErrorState) {
+        attribs[EGL_CONTEXT_FLAGS_KHR] |= EGL_CONTEXT_OPENGL_DEBUG_BIT_KHR;
```

Comment:
The extension and the debug bit are unrelated.
(other than the fact context without the debug bit may opt to generate no log callbacks at all)